### PR TITLE
Create tests for all requirements

### PR DESCRIPTION
**Pull request by NoRage28**

### CODE REVIEW

File: blog/post/tests/test_post.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    def test_like_analytics(self):
+        url = reverse('like-like_analytics')
+        data = {
+            'date_from': '2022-10-25',
+            'date_to': '2022-10-30'
+        }
+        response = self.client.get(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIsInstance(response.json(), list)
+        self.assertEqual(response.json(), [{'likes_count': 1, 'created_at': '2022-10-27'}])
```
Comment (bakarovroman) [inline blog/post/tests/test_post.py:51]: move likes related tests to another tests case. For example: LikesTestCase

File: blog/post/tests/test_post.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    def test_like_create(self):
+        post = Post.objects.create(author=self.user, title='post', content='content')
+        url = reverse('post-like_create', kwargs={'pk': post.pk})
+        response = self.client.post(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(Like.objects.last().author, self.user)
+        self.assertEqual(Like.objects.last().post, post)
+
+    def test_like_remove(self):
```
Comment (bakarovroman) [inline blog/post/tests/test_post.py:36]: test_unlike

File: blog/post/tests/test_post.py (hunk 1 above)
Comment (250kgBeast) [inline blog/post/tests/test_post.py:51]: move likes related tests to different test case(LikeTestCase)

File: blog/post/tests/test_post.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        data = {
+            'title': 'title_text',
+            'content': 'content_text'
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Post.objects.last().title, 'title_text')
+        self.assertEqual(Post.objects.last().content, 'content_text')
+
+    def test_like_create(self):
```
Comment (250kgBeast) [inline blog/post/tests/test_post.py:28]: test_like

File: blog/post/views.py [hunk 4]
```diff
@@ -15,12 +15,12 @@ class PostViewSet(viewsets.ModelViewSet):
     serializer_class = PostSerializer
     permission_classes = [permissions.IsAuthenticatedOrReadOnly]
 
-    @action(detail=True, methods=['post'])
+    @action(detail=True, methods=['post'], url_name='like_create')
```
Comment (250kgBeast) [inline blog/post/views.py:18]: i dont think you need this (`url_name='like_create'`), your url in reverse will be `post-like`

File: blog/post/views.py [hunk 5]
```diff
@@ -15,12 +15,12 @@ class PostViewSet(viewsets.ModelViewSet):
...
     def like(self, request, pk):
         like_post(user=request.user, post_id=pk)
         return Response("Like successfully added")
 
-    @action(detail=True, methods=['post'])
+    @action(detail=True, methods=['post'], url_name='like_remove')
```
Comment (250kgBeast) [inline blog/post/views.py:23]: same as above

File: blog/post/views.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def destroy(self, request, *args, **kwargs):
         unlike_post(user=request.user, post_id=serializer.data['post_id'])
         return Response(status=status.HTTP_204_NO_CONTENT)
 
-    @action(detail=False, methods=['get'], serializer_class=LikeAnalyticsSerializer)
+    @action(detail=False, methods=['get'], serializer_class=LikeAnalyticsSerializer, url_name='like_analytics')
```
Comment (250kgBeast) [inline blog/post/views.py:49]: same as above

File: blog/users/tests/test_authentication.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from django.contrib.auth.models import User
+
+
+class UserSignUpCase(APITestCase):
```
Comment (250kgBeast) [inline blog/users/tests/test_authentication.py:7]: `UserSignUpTestCase`

File: blog/users/tests/test_user_activity.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+@freeze_time('2022-11-01T00:00:00')
+class UserActivityCase(APITestCase):
+
+    def setUp(self) -> None:
+        self.user = User.objects.create_user(username='TestUser', password='1357246max')
+        auth_url = reverse('token_obtain_pair')
+        self.access_token = self.client.post(auth_url, {
+            'username': 'TestUser',
+            'password': '1357246max'
+        }).data.get('access')
```
Comment (250kgBeast) [inline blog/users/tests/test_user_activity.py:18]: move this request to test below

File: blog/users/views.py [hunk 9]
```diff
@@ -18,3 +18,8 @@ class UserActivityViewSet(viewsets.ReadOnlyModelViewSet
                           ):
     queryset = UserActivity.objects.all()
     serializer_class = UserActivitySerializer
+
+    def retrieve(self, request, *args, **kwargs):
+        instance = UserActivity.objects.filter(user_id=self.kwargs.get('pk'))
+        serializer = self.get_serializer(instance, many=True)
+        return Response(serializer.data)
```
Comment (250kgBeast) [inline blog/users/views.py:25]: remove

**Pull request merged**